hummingbird/views.py

```diff
@@ -0,0 +1,195 @@
+from django.shortcuts import render
+from hummingbird.models import UserProfile, UserDevice, User
+from hummingbird.forms import UserProfileForm, UserDeviceForm, UserSongForm
+from django.http import HttpResponse
+import datetime
+import urllib
+
+
+## TO-DO: Add div and context for files in library to allow for file deletion. As of right now, files will just accumulate in the library.
+def index(request):
+    userprofile_list = UserProfile.objects.order_by('-last_played')
+    context_dict = {'userprofiles': userprofile_list}
+    response = render(request, 'index.html', context_dict)
+    return response
+
+
+## Returns info for userprofile page, and will update userprofile/userdevice fields if POST is used.
+def profile(request, user_id):
+	added_song=False
+	added_device=False
+	context_dict = {}
+
+	## If we're getting data via post method, update whatever fields we've been given.
+	if request.method == 'POST':
+		userprofile = UserProfile.objects.get(pk=user_id)
+		usersong_form = UserSongForm(data=request.POST)
+		device_form = UserDeviceForm(data=request.POST)
+	
+		if usersong_form.is_valid():
+			userprofile.song = request.FILES['song']
+			userprofile.length = usersong_form.cleaned_data['length']
+			userprofile.save()
+			added_song = True
+	
+		if device_form.is_valid():
+			device = device_form.save(commit=False)
+			device.user_profile = userprofile
+			added_device = True
+			device.save()
+	
+	## Regardless of method (GET/POST), let's return info for the user's profile.
+	try:
+		userprofile = UserProfile.objects.get(pk=user_id)
+		context_dict['profile'] = userprofile
+		context_dict['name'] = userprofile.name
+		# If user has no song, the template will put in appropriate copy.
+		if userprofile.song:
+			context_dict['song'] = userprofile.song.name
+		# If the user hasn't been played yet, the template will put in appropriate copy.
+		if userprofile.last_played:
+			context_dict['last_played'] = userprofile.last_played.strftime('%Y-%m-%d %H:%M:%S')		
+		context_dict['length'] = userprofile.length
+		# Add in devices to list for user's profile
+		devices = UserDevice.objects.filter(user_profile=userprofile)
+		context_dict['devices'] = devices
+		# Populate forms
+		usersong_form = UserSongForm()
+		device_form = UserDeviceForm()
+		context_dict['usersong_form'] = usersong_form
+		context_dict['device_form'] = device_form
+	except UserProfile.DoesNotExist:
+		pass
+	# If we updated one of the fields, pass that to the template so it can tell the user that update was successful.
+	context_dict['added_device'] = added_device
+	context_dict['added_song'] = added_song
+	return render(request,'profile.html',context_dict)
+
+
+## Checks if user exists. If so, return user name. Otherwise, return "0". 
+def get_user_from_device(request):
+	if request.method == 'GET':
```

Comment:
It's weird that we just pass if the request method is `POST` because Django will throw an exception when a view returns `None`. Maybe we should explicitly throw an exception in the case that the request method is not `GET`, or just ignore the request method entirely and happily pretend it's a `GET` (which could cause unexpected results in rare cases, but I think it's acceptable).